cranelift/codegen/src/isa/aarch64/lower.isle

```diff
@@ -404,7 +404,13 @@
 (rule (lower (has_type ty @ (multi_lane _ _) (fma x y z)))
       (vec_rrr_mod (VecALUModOp.Fmla) z x y (vector_size ty)))
 
-(rule 1 (lower (has_type (ty_scalar_float ty) (fma x y z)))
+(rule 1 (lower (has_type ty @ (multi_lane _ _) (fma (fneg x) y z)))
+      (vec_rrr_mod (VecALUModOp.Fmls) z x y (vector_size ty)))
+
+(rule 2 (lower (has_type ty @ (multi_lane _ _) (fma x (fneg y) z)))
+      (vec_rrr_mod (VecALUModOp.Fmls) z x y (vector_size ty)))
```

Comment:
I suppose if both `x` and `y` are `fneg` then this can emit `fmla` instead of `fneg`+`fmls`, right? But I guess that's a rewrite we ought to do in the egraph optimizations instead.

Reply:
Indeed! The x64 rules actually end up implementing that (they enable sort of switching back and forth given their structure) but it wasn't as obvious to do here - x64 uses a  helper that manages sinking a load as well which adds a fair number of permutations.

I'll send a follow-up which implements the egraph optimization.